packages/create-pantheon-decoupled-kit/src/index.ts

```diff
@@ -85,11 +85,29 @@ export const main = async (
 	// If no generators are found in positional params
 	// ask which generators should be run
 	if (!foundGenerators || !foundGenerators.length) {
-		const whichGenerators: QuestionCollection<{ generators: string[] }> = {
+		// check if a cmsType was provided
+		const cmsType: string | null = args.cmsType
+			? new String(args.cmsType).toLowerCase()
+			: null;
+
+		let suggestedGeneratorNames: string[] | null;
+
+		// filter generators if a cmsType was provided
+		if (cmsType) {
+			const suggestedGenerators = generators.filter((generator) => {
+				return generator.cmsType === cmsType || generator.cmsType === 'any';
+			});
+			suggestedGeneratorNames = suggestedGenerators.map(({ name }) => name);
+		}
+
+		const whichGenerators: QuestionCollection<{
+			generators: string[];
+		}> = {
 			name: 'generators',
 			type: 'checkbox',
 			message: 'Which generator(s) would you like to run?',
-			choices: () => generatorNames,
+			choices: () =>
+				suggestedGeneratorNames ? suggestedGeneratorNames : generatorNames,
```

Comment:
... and this line becomes:
```suggestion
			choices: () =>
				generatorsOfCmsType.length > 0 ? generatorsOfCmsType : generatorNames,
```